src/HasEnums.php

```diff
@@ -16,18 +16,22 @@ trait HasEnums
 {
     public function setAttribute($key, $value)
     {
-        return $this->isEnumAttribute($key)
-            ? $this->setEnumAttribute($key, $value)
-            : parent::setAttribute($key, $value);
+        if (! $this->isEnumAttribute($key) || $this->isNullableEnum($key, $value)) {
```

Comment:
I would like to keep this method as simple as possible and do everything else in the `setEnumAttribute()` method.
It's because this method is a general hook in point for several packages handling attributes and the user shouldn't copy more code than required to combine this with other packages.

Reply:
In conjunction with the `isNullableEnum()` change request this condition should stay like it is and the handling of null values on an enum attribute should be done in `setEnumAttribute()` and not before.